configuration/configuration.schema.json

```
@@ -0,0 +1,143 @@
+          "type": "string"
+        }
+      },
+      "type": "object"
+    },
+    "library": {
+      "description": "configuration options relating to Arduino libraries.",
+      "properties": {
+        "enable_unsafe_install": {
+          "description": "set to `true` to enable the use of the `--git-url` and `--zip-file` flags with [`arduino-cli lib install`][arduino cli lib install]. These are considered \"unsafe\" installation methods because they allow installing files that have not passed through the Library Manager submission process.",
+          "type": "boolean"
+        }
+      },
+      "type": "object"
+    },
+    "locale": {
+      "description": "the language used by Arduino CLI to communicate to the user, the parameter is the language identifier in the standard POSIX format `<language>[_<TERRITORY>[.<encoding>]]` (for example `it` or `it_IT`, or `it_IT.UTF-8`).",
+      "type": "string"
+    },
+    "logging": {
+      "description": "configuration options for Arduino CLI's logs.",
+      "properties": {
+        "file": {
+          "description": "path to the file where logs will be written.",
+          "type": "string"
+        },
+        "format": {
+          "description": "output format for the logs. Allowed values are `text` or `json`.",
+          "type": "string",
+          "enum": ["text", "json"]
+        },
+        "level": {
+          "description": "messages with this level and above will be logged. Valid levels are: `trace`, `debug`, `info`, `warn`, `error`, `fatal`, `panic`.",
+          "type": "string",
+          "enum": ["trace", "debug", "info", "warn", "error", "fatal"]
+        }
+      },
+      "type": "object"
+    },
+    "metrics": {
+      "description": "settings related to the collection of data used for continued improvement of Arduino CLI.",
+      "properties": {
+        "addr": {
+          "description": "TCP port used for metrics communication.",
+          "type": "string"
+        },
+        "enabled": {
+          "description": "controls the use of metrics.",
+          "type": "boolean"
+        }
+      },
+      "type": "object"
+    },
+    "sketch": {
+      "description": "configuration options relating to [Arduino sketches][sketch specification].",
+      "properties": {
+        "always_export_binaries": {
+          "description": "set to `true` to make [`arduino-cli compile`][arduino-cli compile] always save binaries to the sketch folder. This is the equivalent of using the [`--export-binaries`][arduino-cli compile options] flag.",
+          "type": "boolean"
+        }
+      },
+      "type": "object"
+    },
+    "updater": {
+      "description": "configuration options related to Arduino CLI updates",
+      "properties": {
+        "enable_notification": {
+          "description": "set to `false` to disable notifications of new Arduino CLI releases, defaults to `true`",
+          "type": "boolean"
```

```suggestion
          "type": "boolean",
          "default": true
```

https://json-schema.org/understanding-json-schema/reference/generic.html#annotations

Even though it is not used in validation, it seems like defaults would be worth adding to provide some extra value in case the schema is used to enhance the configuration file editing process (e.g., in VS Code) and they might also be used in the documentation generated from the schema.